src/files/glob-source.js

```diff
@@ -7,131 +7,71 @@ const Path = require('path')
 const errCode = require('err-code')
 
 /**
- * Create an async iterator that yields paths that match requested file paths.
+ * Create an async iterator that yields paths that match requested glob pattern
  *
- * @param {Iterable<string> | AsyncIterable<string> | string} paths - File system path(s) to glob from
+ * @param {string} pattern - Glob pattern to match
  * @param {Object} [options] - Optional options
- * @param {boolean} [options.recursive] - Recursively glob all paths in directories
+ * @param {string} [options.cwd] - The directory to start matching the pattern in
  * @param {boolean} [options.hidden] - Include .dot files in matched paths
- * @param {Array<string>} [options.ignore] - Glob paths to ignore
  * @param {boolean} [options.followSymlinks] - follow symlinks
  * @param {boolean} [options.preserveMode] - preserve mode
  * @param {boolean} [options.preserveMtime] - preserve mtime
  * @param {number} [options.mode] - mode to use - if preserveMode is true this will be ignored
  * @param {import('ipfs-unixfs').MtimeLike} [options.mtime] - mtime to use - if preserveMtime is true this will be ignored
  * @yields {Object} File objects in the form `{ path: String, content: AsyncIterator<Buffer> }`
  */
-module.exports = async function * globSource (paths, options) {
+module.exports = async function * globSource (pattern, options) {
```

Comment:
I'd be tempted to make this take `cwd, pattern, options` to avoid some of the cwd messing around bits.

Reply:
Done, I'm, always wondering where the output path will start.